cmd/fleet/actionDispatcher.go

```diff
@@ -0,0 +1,120 @@
+package fleet
+
+import (
+	"context"
+	"fleet/internal/pkg/action"
+	"fleet/internal/pkg/esutil"
+	"fleet/internal/pkg/seqno"
+	"sync"
+
+	"github.com/rs/zerolog/log"
+)
+
+type ActionX = action.ActionX
+
+type ActionSubX struct {
+	agentId string
+	seqNo   int64
+	ch      chan []ActionX
+}
+
+func (as ActionSubX) Ch() chan []ActionX {
+	return as.ch
+}
+
+type ActionDispatcher struct {
+	am *seqno.Monitor
+
+	mx   sync.RWMutex
+	subs map[string]ActionSubX
+}
+
+func NewActionDispatcher(am *seqno.Monitor) *ActionDispatcher {
+	return &ActionDispatcher{
+		am:   am,
+		subs: make(map[string]ActionSubX),
+	}
+}
+
+func (ad *ActionDispatcher) Run(ctx context.Context) (err error) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case hits := <-ad.am.Output():
+			ad.process(ctx, hits)
+		}
+	}
+}
+
+func (ad *ActionDispatcher) Subscribe(agentId string, seqNo int64) *ActionSubX {
+	cbCh := make(chan []ActionX, 1)
+
+	sub := ActionSubX{
+		agentId: agentId,
+		seqNo:   seqNo,
+		ch:      cbCh,
+	}
+
+	ad.mx.Lock()
+	ad.subs[agentId] = sub
+	sz := len(ad.subs)
+	ad.mx.Unlock()
+
+	log.Trace().Str("agentId", agentId).Int("sz", sz).Msg("Action dispatcher subscribe")
+
+	return &sub
+}
+
+func (ad *ActionDispatcher) Unsubscribe(sub *ActionSubX) {
+	if sub == nil {
+		return
+	}
+
+	ad.mx.Lock()
+	delete(ad.subs, sub.agentId)
+	sz := len(ad.subs)
+	ad.mx.Unlock()
+
+	log.Debug().Str("agentId", sub.agentId).Int("sz", sz).Msg("Action dispatcher unsubscribe")
+}
+
+func (ad *ActionDispatcher) process(ctx context.Context, hits []esutil.Hit) {
+	// Parse hits into map of agent -> actions
+	// Actions are ordered by sequence
+
+	actions := action.HitsToActions(hits)
+
+	agentActions := make(map[string][]ActionX)
+	for _, action := range actions {
+		for _, agentId := range action.Agents {
+			arr := agentActions[agentId]
+			arr = append(arr, action)
+			agentActions[agentId] = arr
+		}
+	}
+
+	// TODO: revisit actions rollout
```

Comment:
Whats the TODO here? Would be good to comment on the improvement you think should be made here.